src/impl/index.ts

```diff
@@ -0,0 +1,106 @@
+import { ZoneContextManager } from '@opentelemetry/context-zone';
+import { ALWAYS_SAMPLER, B3Propagator, NEVER_SAMPLER, ProbabilitySampler } from '@opentelemetry/core';
+import { CollectorExporter, CollectorExporterConfig } from '@opentelemetry/exporter-collector';
+import { XMLHttpRequestPlugin } from '@opentelemetry/plugin-xml-http-request';
+import { PropagateTraceHeaderCorsUrls } from '@opentelemetry/plugin-xml-http-request/build/src/types';
+import { ConsoleSpanExporter, SimpleSpanProcessor } from '@opentelemetry/tracing';
+import { WebTracerProvider } from '@opentelemetry/web';
+
+export interface TracingProperties {
+  samplingRate: number;
+  corsUrls: PropagateTraceHeaderCorsUrls;
+  collectorConfiguration: CollectorExporterConfig | undefined;
+  consoleOnly: boolean;
+}
+
+/**
+ * TODOs:
+ * - other provider config options via props
+ * - allow propagator definition via props
+ */
+export default class OpenTelemetryTracingImpl {
+  private props: TracingProperties;
+  private beaconUrl: string;
+  private initialized: boolean;
+
+  constructor() {
+    // set default properties
+    this.props = {
+      samplingRate: 1,
+      corsUrls: [],
+      collectorConfiguration: undefined,
+      consoleOnly: false,
+    };
+    this.initialized = false;
+  }
+
+  public register = () => {
+    if (this.initialized) {
+      return;
+    }
+
+    // provider with the only one available plugin for now
+    const provider = new WebTracerProvider({
+      sampler: this.resolveSampler(),
+      plugins: [
+        new XMLHttpRequestPlugin({
+          propagateTraceHeaderCorsUrls: this.props.corsUrls,
+        }),
+      ],
+    });
+
+    // if we have any exporter available, then any of them, otherwise use the console logger
```

Comment:
This comment is wrong because it is not checking if any exporter is available and using any of them.